### DISCUSSION TITLE
multiple relation between 2 tables and automatic lines

### DISCUSSION THREAD

**Discussion by Daybayzayd** [dineug/erd-editor#281] [Ideas]
Hello,

First, your work is very beautiful and easy to use ! Very thanks.

I have 2 questions, maybe they are already posted.

1/ Can we have multiple relation between 2 tables ? Between 1 table and x other tables ?

For exemple like this :

![diagram_1](https://user-images.githubusercontent.com/55182867/139154332-236a4a8a-3eb5-4a71-8abf-91089a66ea98.png)

2/ Can we have the line not across the table but "near" the table like this example :

![diagram_2](https://user-images.githubusercontent.com/55182867/139154381-8958cf05-4238-426b-9896-7a4be1b93d49.png)

We have model with 20 or 30 tables, and with multiple relation and we want something like this :

![Sales-Tracker-DbDesigner-1080x675](https://user-images.githubusercontent.com/55182867/139154684-bb499ff5-f04d-4b4f-b29c-0991c91a6594.png)

Is this possible actually ?

**Comment by dineug** [comment]
Using [PathFinding.js](https://github.com/qiao/PathFinding.js), it seems possible to implement it.
I'll add it to my worklist.

**Comment by Daybayzayd** [comment]
Very nice, the unique problem i see is maybe that the project is old. :(

I don't know if you know this project, but i would like something like this :

https://dbdiagram.io/d

My projet is to transform my xml with table and relation into a json format to show an ER Diagram like yours dynamically. :)

**Comment by dineug** [comment]
Update v0.9.12 and try again.

**Comment by Daybayzayd** [comment]
Hello, Thanks for your work.

It doesn't seem to work properly ?
It was working weirdly when i tested it 1 week ago, but now it seems to not working..
![vuerd_pathfinding](https://user-images.githubusercontent.com/55182867/141506547-b9c45694-765a-460f-848d-4796304c9f6a.PNG)

https://codepen.io/dineug/embed/jOyoJYR?height=274&theme-id=dark&default-tab=js,result

**Comment by dineug** [comment]
It can be activated in settings.
The smaller the canvas size, the better it works.
There's a performance issue, so this is the best I can do right now.

<img width="622" alt="스크린샷 2021-11-13 오전 2 16 25" src="https://user-images.githubusercontent.com/45829489/141507823-fa9f7eff-dae1-4c45-bbb5-f2673d0f3479.png">
